src/dsp_tools/utils/excel_to_json/properties.py

```diff
@@ -0,0 +1,457 @@
+    # These are also mandatory but require formatting
+    _property.update(
+        {"labels": utl.get_labels(df_row=prop_row), "super": [s.strip() for s in prop_row["super"].split(",")]}
+    )
+    non_mandatory = {
+        "comments": utl.get_comments(df_row=prop_row),
+        "gui_attributes": _get_gui_attribute(df_row=prop_row, row_num=row_count, excel_filename=excel_filename),
+    }
+    # These functions may return None, this is checked before the update
+    _property = utl.update_dict_ifnot_value_none(additional_dict=non_mandatory, to_update_dict=_property)
+    return _property
+
+
+def _check_gui_attributes(check_df: pd.DataFrame) -> dict[str, pd.Series] | None:
+    """
+    This function takes a pd.DataFrame and checks if the "gui_attributes" column is filled correctly. If any
+    or all of the checks fail, it creates a dictionary with a pd.Series as value which contains True for all rows where
+    there is a problem otherwise, it returns None.
+
+    Args:
+        check_df: pd.DataFrame that should be checked
+
+    Returns:
+        A dictionary with a pd.Series that contains the information where there is a problem or None if all the
+        checks passed.
+
+    Raises:
+        UserError if any of the checks fail
+    """
+    mandatory_attributes = ["Spinbox", "List"]
+    mandatory_check = utl.col_must_or_not_empty_based_on_other_col(
+        check_df=check_df,
+        substring_list=mandatory_attributes,
+        substring_colname="gui_element",
+        check_empty_colname="gui_attributes",
+        must_have_value=True,
+    )
+    no_attributes = ["Checkbox", "Date", "Geonames", "Richtext", "TimeStamp"]
+    no_attribute_check = utl.col_must_or_not_empty_based_on_other_col(
+        check_df=check_df,
+        substring_list=no_attributes,
+        substring_colname="gui_element",
+        check_empty_colname="gui_attributes",
+        must_have_value=False,
+    )
+    # If neither has a problem, we return None
+    if mandatory_check is None and no_attribute_check is None:
+        return None
+    # If both have problems, we combine the series
+    elif mandatory_check is not None and no_attribute_check is not None:
+        final_series = pd.Series(np.logical_or(mandatory_check, no_attribute_check))
+    elif mandatory_check is not None:
+        final_series = mandatory_check
+    else:
+        final_series = no_attribute_check
+    # The boolean series is returned
+    return {"wrong gui_attributes": final_series}
+
+
+def _check_missing_values_in_row_raise_error(to_check_df: pd.DataFrame, excel_filename: str) -> None:
+    """
+    This function checks if all the required values are in the df. If all the checks are ok, the function ends without
+    any effect. If any of the checks fail, the called functions return a dictionary with the column name as key,
+    After all the checks are done, the a UserError is raised which contains the information in which column and row
+    there are problems.
```

Comment:
"After all the checks are done, the a UserError is raised":

- "the a" must be a typo
- this sounds as if the error was always raised after all checks are done, which is not true